nevergrad/optimization/optimizerlib.py

```diff
@@ -866,79 +877,78 @@ def __init__(
             num_workers: int = 1,
             num_optims: tp.Optional[int] = None,
             num_vars: tp.Optional[tp.List[int]] = None,
-            multivariate_optimizer: base.ConfiguredOptimizer = CMA,
-            monovariate_optimizer: base.ConfiguredOptimizer = RandomSearch,
+            multivariate_optimizer: base.OptCls = CMA,
+            monovariate_optimizer: base.OptCls = RandomSearch,
             progressive: bool = False,
     ) -> None:
         super().__init__(parametrization, budget=budget, num_workers=num_workers)
+        self._subcandidates: tp.Dict[str, tp.List[p.Parameter]] = {}
+        self._progressive = progressive
+        subparams: tp.List[p.Parameter] = []
         if num_vars is not None:  # The user has specified how are the splits (s)he wants.
-            if num_optims is not None:  # (S)he also specifies the number of splits.
-                assert num_optims == len(num_vars), f"The number {num_optims} of optimizers should match len(num_vars)={len(num_vars)}."
-            else:  # Better: we deduce the number of splits.
-                num_optims = len(num_vars)
             assert sum(num_vars) == self.dimension, f"sum(num_vars)={sum(num_vars)} should be equal to the dimension {self.dimension}."
-        else:  # The user did not specify the number of vars per split.
-            if num_optims is None:  # if no num_vars and no num_optims, try to guess how to split. Otherwise, just assume 2.
-                if isinstance(parametrization, p.Parameter):
-                    param_val = [x[1] for x in sorted(parametrization.value.items(), key=lambda x: int(x[0]))]
-                    num_vars = []
-                    for param_v in param_val:
-                        num_vars += [param_v.dimension if isinstance(param_v, p.Parameter) else 1]
-                    num_optims = len(num_vars)
-                else:  # Desperate situation: just split in 2.
-                    num_optims = 2
+            if num_optims is None:  # we deduce the number of splits.
+                num_optims = len(num_vars)
+            assert num_optims == len(num_vars), f"The number {num_optims} of optimizers should match len(num_vars)={len(num_vars)}."
+        elif num_optims is None:
+            # if no num_vars and no num_optims, try to guess how to split. Otherwise, just assume 2.
+            if isinstance(parametrization, p.Parameter):
+                subparams = [x[1] for x in paramhelpers.split_as_data_parameters(parametrization)]
```

Comment:
@teytaud should we add a `noisy` tag to all parametrizations? Because splitting makes the problem noisy